lib/puma/server.rb

```diff
@@ -589,18 +589,17 @@ def handle_request(req, lines)
 
             return :async
           end
-        rescue ThreadPool::ForceShutdown => e
-          @events.log "Detected force shutdown of a thread, returning 503"
-          @events.unknown_error self, e, "Rack app"
-
-          status = 503
-          headers = {}
-          res_body = ["Request was internally terminated early\n"]
-
         rescue Exception => e
           @events.unknown_error self, e, "Rack app", env
 
-          status, headers, res_body = lowlevel_error(e, env)
+          status = 500
+          if e.is_a?(ThreadPool::ForceShutdown)
```

Comment:
I think this is less clear/nice looking than what it was before. Can we just keep the separate handling for ThreadPool::ForceShutdown (it's own rescue block) as it was before?

Reply:
Sure, I changed it back.